clang/utils/TableGen/ClangDiagnosticsEmitter.cpp

```diff
@@ -68,7 +69,8 @@ getCategoryFromDiagGroup(const Record *Group,
 
   // The diag group may the subgroup of one or more other diagnostic groups,
   // check these for a category as well.
-  const std::vector<Record*> &Parents = DiagGroupParents.getParents(Group);
+  const std::vector<const Record *> &Parents =
```

Comment:
Should we change this to use an `ArrayRef`? (Same for the `getParents()` API?)

Reply:
We could, I was trying to not change more than what's necessary for this change. I think there are other cleanup like this that can done for all the clang tablegen backends (like using ArrayRef instead of const ref to vectors, range based for loops etc). I was thinking that those can be done in a separate NFC (I can do that once this series of patches goes in).

Reply:
Sure, that works for me, thank you!